src/components/Portfolio.jsx

```diff
@@ -1,170 +1,67 @@
-              
-            </small>
-          </Card.Footer>
-        </Card>
-
-{/* CARD 02  */}
-
-        <Card>
-          <Card.Img className='img-portfolio' variant="top" src={Password} />
-          <Card.Body className='card-portfolio'>
-            <Card.Title><a
-          href="https://inaiak.github.io/Password-generator/"
-          className="event-link"
-          target="_blank"
-          rel="noreferrer"
-        >
-          {" "}Password Generator{" "}
-        </a></Card.Title>
-            <Card.Text>
-            An application that enables employees to generate random passwords based on criteria that they’ve selected. 
-            This app will run in the browser and will feature dynamically updated HTML and CSS powered by JavaScript. It will have a clean and polished, responsive user interface that adapts to multiple screen sizes.{' '}
-            </Card.Text>
-          </Card.Body>
-          <Card.Footer>
-            <small className="text-muted">Technologies:HTML, CSS, JavaScript</small>
-          </Card.Footer>
-        </Card>
-
-        {/* CARD 03 */}
-        <Card>
-          <Card.Img className='img-portfolio' variant="top" src={Note} />
-          <Card.Body className='card-portfolio'>
-            <Card.Title><a
-          href="https://tranquil-dawn-45157.herokuapp.com/"
-          className="event-link"
-          target="_blank"
-          rel="noreferrer"
-        >
-          {" "}Note Taker {" "}
-        </a></Card.Title>
-            <Card.Text>
-            Note Taker that can be used to write and save notes.
-                       </Card.Text>
-          </Card.Body>
-          <Card.Footer>
-            <small className="text-muted">Technologies:Bootstrap, JavaScript, Express.js</small>
-          </Card.Footer>
-        </Card>
-        {/* CARD 04  */}
-
-        <Card>
-          <Card.Img className='img-portfolio' variant="top" src={Coming} />
-          <Card.Body>
-            <Card.Title>Future Project</Card.Title>
-            <Card.Text>
-            </Card.Text>
-          </Card.Body>
-          <Card.Footer>
-            <small className="text-muted">Technologies</small>
-          </Card.Footer>
-        </Card>       
-      </CardGroup>
+  const renderCard = (card, index) => {
+    return (
+      <Card style={{ width: "20rem" }} key={index} className="card.box">
+        <Card.Img variant="top" src={card.img} alt={card.altText} />
+        <Card.Body>
+          <Card.Title>{card.title}</Card.Title>
+          <Card.Text>{card.description}</Card.Text>
+          <Button>
+            <a href={card.deployed} target="_blank" rel="noreferrer noopener">
+              Visit {card.title}
+            </a>
+          </Button>
+        </Card.Body>
+      </Card>
     );
-  }
-  
-export default Portfolio;  
-
-
-
-
-
+  };
 
-//plan b 
+  return <div className="grid"> {cardData.map(renderCard)}</div>;

```

Comment:
nice! defining the function in advance and then passing it into `.map()` is a fairly advanced pattern that isn't easy for some folks to grasp--great work 👏